content/doc/book/using/aborting-a-build

```diff
@@ -0,0 +1,62 @@
+---
+layout: section
+title: Aborting a build
+---
+
+= Aborting a build
+
+When you click that [x] icon from the UI, the following things happen:
+
+. Browser sends a request to the server.
+. The server interrupts (via Thread.interrupt()) the thread (AKA
+executor thread) that is responsible for carrying out a build.
+. The server returns
+
+At this point, your browser is back, but the actual abort process
+happens from here asynchronously.
+
+. The thread gets the interrupt signal. How quickly this happens depends
+on what the executor is doing at the time of the interruption.
+Specifically, an executor thread can only be interrupted in
+"interruption points" due to the Java design.
+* Waiting for a completion of a child process (for example, maybe the
+build is running Ant) is an interruption point. That means if the
+executor was doing that, it gets interrupted instantaneously.
+* Waiting for a computation on a slave is an interruption point.
+* Waiting for file or network I/O is not an interruption point. This
+often causes the problem where a build appears to be un-abortable. For
+example, checking out a Subversion repository falls in this category.
+* Normal computation is also not an interruption point.
+. The executor performs a clean up operation. This depends on what it
+was doing by the time it noticed the interruption.
+* If it was waiting for a completion of a child process, Jenkins will
+search for all the descendant processes and kill them all. On Unix, this
+is done through `+java.lang.UnixProcess.destroyProcess+`, which sends
+http://en.wikipedia.org/wiki/SIGTERM[SIGTERM] on Sun's JREs. On Windows,
+this is done through
+http://msdn.microsoft.com/en-us/library/ms686714(VS.85).aspx[TerminateProcess]
+API.
+* If it was waiting for a completion of some computation in a agent, the
+thread that's performing the remote computation is interrupted
+asynchronously. How quickly that threads gets interrupted depends on
+what that thread is doing. See above.
+. Executor starts unwinding the stack, and eventually it finishes the
+unwinding. At this point, the build is marked as aborted and executor
+gets back to the idle status.
```

Comment:
Consider avoiding phrasal verbs to make it easier to understand for non-native English readers.  The English verb 'get' is too flexible for its own good.
```suggestion
returns to the idle status.
```